app/dts/bindings/zmk,led-indicators.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2021 The ZMK Contributors
+# SPDX-License-Identifier: MIT
+
+description: LED indicators
+
+compatible: "zmk,led-indicators"
```

Comment:
Is there a reason not to use the `gpio-leds` bindings and just leverage that?

Seems like we could do:

```
/ {
  chosen {
    zmk,led-indicators = &indicators;
  };

  indicators: gpio_indicators {
    compatible = "gpio-leds";
      led_numlock {
        gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
     };
  };
};
```

And leverage it directly. Check out `app/src/backlight.c` for an example where we are already using this.

Reply:
There are two downsides with your proposal:

* How to find which indicator corresponds to a LED? We can use the child name, as in your example, but then we'll have to find a way to translate the name from a string to the indicator. This approach also doesn't allow multiple LEDs with the same indicator.
* This approach works for LED_GPIO and LED_PWM drivers, but it doesn't work with other drivers, however I guess very few users have other drivers.

If you think this approach is better than the current one, I can change it.

Reply:
I like this approach and I agree that having multiple LEDs for the same indicator is not very useful.

Reply:
I would suggest we go this route then, for the DTS side of things.